src/webgpu/api/validation/image_copy/buffer_related.spec.ts

```diff
@@ -148,38 +148,48 @@ g.test('bytes_per_row_alignment')
 Test that bytesPerRow must be a multiple of 256 for CopyB2T and CopyT2B if it is required.
 - for all copy methods between linear data and textures
 - for all texture dimensions
-- for all sized formats.
-- for various bytesPerRow aligned to 256 or not
-- for various number of blocks rows copied
+- for all sized (format,aspect) pairs, except for a bunch of redundant "regular" color formats
+- for various bytesPerRow (aligned to 256 or not)
+- for various number of blocks rows copied (0 or not)
 `
   )
   .params(u =>
     u //
       .combine('method', kImageCopyTypes)
-      .combine('format', kSizedTextureFormats)
-      .filter(formatCopyableWithMethod)
       .combine('dimension', kTextureDimensions)
+      .combine('format', kReducedFormatsList)
+      .expandWithParams(p => aspectParamsForTextureFormat(p.format))
+      .filter(p => aspectCopyableWithMethod(p.format, p._aspectKey, p.method))
       .filter(({ dimension, format }) => textureDimensionAndFormatCompatible(dimension, format))
       .beginSubcases()
-      .combine('bytesPerRow', [undefined, 0, 1, 255, 256, 257, 512])
-      .combine('copyHeightInBlocks', [0, 1, 2, 3])
+      .combineWithParams([
+        // valid: height<=1
+        { copyHeightInBlocks: 0, bytesPerRow: 1 },
+        { copyHeightInBlocks: 0, bytesPerRow: undefined },
+        { copyHeightInBlocks: 1, bytesPerRow: 1 },
+        { copyHeightInBlocks: 1, bytesPerRow: undefined },
+        // valid: multiples of 256
+        { copyHeightInBlocks: 2, bytesPerRow: 0 },
+        { copyHeightInBlocks: 2, bytesPerRow: 512 },
+        { copyHeightInBlocks: 2, bytesPerRow: 256 },
+        // valid only for writeTexture
+        { copyHeightInBlocks: 2, bytesPerRow: 128 },
+        { copyHeightInBlocks: 2, bytesPerRow: 1 }, // with 1-byte formats
+        { copyHeightInBlocks: 2, bytesPerRow: 2 }, // with <=2-byte formats
+        { copyHeightInBlocks: 2, bytesPerRow: 8 }, // with <=8-byte formats
+        { copyHeightInBlocks: 2, bytesPerRow: 32 }, // with <=32-byte formats
+      ])
       .expand('_textureHeightInBlocks', p => [
         p.copyHeightInBlocks === 0 ? 1 : p.copyHeightInBlocks,
       ])
       .unless(p => p.dimension === '1d' && p.copyHeightInBlocks > 1)
       // Depth/stencil format copies must copy the whole subresource.
-      .unless(p => {
-        const info = kTextureFormatInfo[p.format];
-        return (
-          (!!info.depth || !!info.stencil) && p.copyHeightInBlocks !== p._textureHeightInBlocks
-        );
-      })
+      .unless(p => p._aspectKey !== 'color' && p.copyHeightInBlocks !== p._textureHeightInBlocks)
```

Comment:
The only `p.copyHeightInBlocks !== p._textureHeightInBlocks` case is `p.copyHeightInBlocks = 0, p._textureHeightInBlocks = 1` which can be a no-op. I think it's better expand `_textureHeightInBlocks` with `[p.copyHeightInBlocks, pp.copyHeightInBlocks + 1]` for > 0 cases?